core/model/modx/modrequest.class.php

```diff
@@ -197,11 +197,6 @@ public function getResource($method, $identifier, array $options = array()) {
                 $resource->fromArray($cachedResource['resource'], '', true, true, true);
                 $resource->_content = $cachedResource['resource']['_content'];
                 $resource->_isForward = $isForward;
-                if (isset($cachedResource['contentType'])) {
```

Comment:
I don't think this is right. The cached resource is using this to restore the object.

Reply:
Yes, in a case when it was an old cache and need to add the proper modContentType object to `content_type` field. I don't think that anywhere exists so old site with so old cache where `content_type` field doesn't exist.

Reply:
It's not old cache. This is how the new ContentType objects are loaded from cache. Without it, the ContentType object is not available from what I can see.

Reply:
I think this needs to be refactored so that the ContentType is loaded from cache in a different way?

Reply:
@Alroniks It looks like you took out the contentType out of the cache in modCacheManager->generateResource, and because of that also removed it from here - any particular reason for that? I suspect that could be a (small) performance improvement to keep it in.